tap_bigcommerce/bigcommerce.py

```diff
@@ -293,7 +295,7 @@ def _response_hook(self, resp, *args, **kwargs):
             elif resp.status_code == 429:
                 raise BigCommerceRateLimitException(resp)
             else:
-                raise HTTPError(resp)
+                raise BigCommerceServerUnresponsive(resp)
```

Comment:
I think that we should only raise this for now if it's a 500->505 error? Currently this will raise on any code that's not 204 or 429, which would include 404s etc